shards/shards.go

```diff
@@ -667,13 +667,71 @@ search:
 			r.Priority = r.priority
 			r.MaxPendingPriority = pending.max()
 
-			sender.Send(r.SearchResult)
+			// For simple shards, send the results immediately.
+			if len(r.SearchResult.RepoURLs) <= 1 {
+				sender.Send(r.SearchResult)
+				break
+			}
+
+			// For compound shards, stream results per repo.
+			stats, srs := splitByRepository(r.SearchResult)
+			tr.LazyPrintf("compound shard: results from %d repositories", len(srs))
+			for _, sr := range srs {
+				if len(sr.Files) == 0 {
+					panic("unreachable")
+				}
+				sr.Priority = sr.Files[0].RepositoryPriority
+
+				// TODO (stefan): make sure repos in compound shards are searched in order of their priority.
```

Comment:
Leftover?

Reply:
That is an acutall TODO which is handled in #197, which reminds me I should take out the TODO there :-). I left the TODO in because sending the MaxPendingPriority like this is actually wrong if repos in compound shards are not sorted.